### ISSUE TITLE
CS231N 7강 - Training Neural Networks II

### ISSUE THREAD

**Issue by dxdiag20** [IPCVLabEdu/flipped_learning#28]
1. SGD가 수직에 민감하다는데 이해가 안갑니다. 수평보다는 수직으로 움직여서 수직에만 더 민감하게 반응하는 건가요? 그렇다면 수직으로 움직인다는게 그래프상에서 Y축 방향으로 계산식이 집중 되있어서 그런가요? 최적화에서 방향에 대해서 추가적인 이해가 필요한것 같습니다. 그리고 SGD가 실제 쓰이기는 하나요?
2. flat minima 와 sharp minima의 차이 sharp minima가 실제로 테스트에서 성능이 떨어지나요?
3. 최적화 방식은 무조건 Adam을 쓰는게 좋은건지 아니면 모델마다 적합한 최적화 방식이 있는건지 궁금합니다.
4. Single model이라는 것은 무엇인지?
5. Dropout은 FC층에 적용시키는게 나을까요? 아니면은 먼저 그냥 돌려보고 과소적합 생기면 레이어 중간중간에 넣는 것이 나을까요?

**Comment by jungin500** [comment]
# 20190221 토의 내용

![image](https://user-images.githubusercontent.com/5201073/53160955-57ad6680-360c-11e9-8c60-20d40a606338.png)

1. 위 강의자료 pp. Lect7-22에 나온 예를 보면, 해당 그래프는 수직 축 가중치(임의로 $W_1$)와 수평 축 가중치 $W_2$ 두개에 대해서 존재합니다. 여기서는 수직 축 가중치 $W_1$가 수평 축 가중치 $W_2$보다 더 민감하게 변화하는데, 여기서 보여주고자 하는 것은 "**여러 가중치가 존재하는데, 그중 어떤 가중치(여기서는 $W_2$)는 더 둔하게 반응하고, 어떤 가중치($W_1$)는 더 민감하게 반응할 수 있다**" 라는 것을 알 수 있습니다.

![image](https://user-images.githubusercontent.com/5201073/53161713-db1b8780-360d-11e9-914e-18df6162178d.png)

2. Flat minima는 Convex한(복잡한) loss 함수에서의 평평한 부분을, Sharp minima는 위 강의자료 pp. Lecture 7-29 에서 보이는 Local minima의 부분을 의미합니다.

3. `Learning rate`는 현재 상황과는 다르게 얼마나 갈 지 (보폭)를 결정하는 하이퍼페러미터이며, `Momentum`의 경우 현재 상황에 따라 변화합니다. 결론: 현재로서는 Adam을 주로 많이 사용하지만, 복잡하지 않은 문제의 경우 SGD를 많이 사용합니다.

4. 입력부터 출력까지 중간 모델이 단일(하나)인 것을 의미합니다.

5. 직접 사용해봐야 알 수 있을 것이라고 합니다.
